### PR TITLE
watchers: use resource for network policies

### PR DESCRIPTION
**Pull request by bimmlerd**
Transform network policy store and informer to the resource pattern.

### CODE REVIEW

File: pkg/k8s/resource_ctors.go [hunk 1]
```diff
@@ -107,6 +108,17 @@ func CiliumIdentityResource(lc hive.Lifecycle, cs client.Clientset, opts ...func
...
 
+func NetworkPolicyResource(lc hive.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) (resource.Resource[*slim_networkingv1.NetworkPolicy], error) {
+	if !cs.IsEnabled() {
+		return nil, nil
+	}
+	lw := utils.ListerWatcherWithModifiers(
+		utils.ListerWatcherFromTyped[*slim_networkingv1.NetworkPolicyList](cs.Slim().NetworkingV1().NetworkPolicies("")),
+		opts...,
+	)
+	return resource.New[*slim_networkingv1.NetworkPolicy](lc, lw, resource.WithMetric("NetworkPolicyList")), nil
```
Comment (joamaki) [inline pkg/k8s/resource_ctors.go]: Metric should be just "NetworkPolicy"

**Review by joamaki — APPROVED** [review]

File: pkg/k8s/resource_ctors.go (hunk 1 above)
Comment (bimmlerd) [reply pkg/k8s/resource_ctors.go]: Ack - should the above `CiliumIdentityList` then also be just `CiliumIdentity`?

File: pkg/k8s/watchers/network_policy.go [hunk 2]
```diff
@@ -4,76 +4,65 @@
...
+		case event, ok := <-events:
+			if !ok {
+				return
+			}
+			var err error
+			switch event.Kind {
+			case resource.Sync:
+				var store resource.Store[*slim_networkingv1.NetworkPolicy]
+				synced.Store(true)
+				store, err = k.resources.NetworkPolicies.Store(ctx)
```
Comment (squeed) [inline pkg/k8s/watchers/network_policy.go:53]: is it safe to close `networkPoliciesStoreSet` if err is non-nil here?

Comment (bimmlerd) [reply pkg/k8s/watchers/network_policy.go:53]: ugh, no that's an oversight. Thanks!

**Review by squeed — APPROVED** [review]

**Comment by bimmlerd** [comment]
CI triage:
- ci-awscni: fatal error in cilium-cli, concurrent map access. Filed https://github.com/cilium/cilium-cli/issues/1794.
- ci-multicluster failed twice; [cilium-sysdumps.zip](https://github.com/cilium/cilium/files/11965965/cilium-sysdumps.zip):
   - "[Installation and Connectivity Test (1, disabled, ipv4, disabled, iptables, true, legacy, legacy)](https://github.com/cilium/cilium/actions/runs/5455131938/jobs/9926185005#logs)" with `connectivity test failed: timeout reached waiting lookup for localhost from pod cilium-test/client-6965d549d5-7vrh4 to server on pod cilium-test/echo-other-node-545c9b778b-8d8w6 to succeed (last error: context deadline exceeded)` seems like https://github.com/cilium/cilium/issues/26513.
   - [Installation and Connectivity Test (7, vxlan, ipv4, wireguard, iptables, true, cluster, cluster)](https://github.com/cilium/cilium/actions/runs/5455131938/jobs/9926185905#logs) also failed with ` connectivity test failed: timeout reached waiting lookup for localhost from pod cilium-test/client-6965d549d5-jjs5w to server on pod cilium-test/echo-other-node-545c9b778b-wjq9b to succeed (last error: context deadline exceeded)`, again like https://github.com/cilium/cilium/issues/26513.

Both seem unrelated/known flakes, rerunning.

**Comment by bimmlerd** [comment]
CI triage:
- Travis CI hit: https://github.com/cilium/cilium/issues/26617

**Comment by bimmlerd** [comment]
CI is green, approvals are in, marking this ready-to-merge. (Why didn't MLH? I guess because I reran the Travis suite via UI instead of command?)

**Pull request merged**